cmd/query/app/static_handler.go

```diff
@@ -149,16 +149,17 @@ func (sH *StaticAssetsHandler) watch() {
 					}
 					continue
 				}
+				if event.Op&fsnotify.Write == fsnotify.Write || event.Op&fsnotify.Create == fsnotify.Create {
```

Comment:
1. reduce nesting by inverting the condition and calling `continue`, as above
2. should this be checking `event.Name` to match the actual config file, rather than reacting to all changes in the directory?
3. another option to reduce logging is to compare the file context before and after, and only log on changed content. This is the approach taken for reloading sampling strategies file

Reply:
1) Ok, sounds reasonable.

2) I agree , but at the same time I'm not sure why in https://github.com/jaegertracing/jaeger/pull/2254/files#diff-5893b37f4eb854b962cce2c3be922456R180 we watch directory changes too, that line gave me the impression that we want to react to directory changes , but not sure why and I might be misunderstanding something here.

3) Compare file content, we can do that, or we can compute the md5sum or something like that. but Do we really need that? The only scenario which I observed the excessive logging is due `CHMOD` events, of course can be other scenarios and a comparison could prevent future issues with like this one.

Reply:
2 - I remember something about renaming in k8s, etc. which required watching the directory. And when watching the directory, there could be many other changes happening, although it's still strange that the user experience logs every 0.5s.  By checking that the event is actually for the file name we want we can eliminate all other events in the dir.

3 - we can try with just adding the name check, although doing a full comparison on `(string)content` is trivial, just a couple of lines right in this function.

Reply:
I did all except the content comparison, I don't think we need it at this point but if you think is good I can add it, as you said is a trivial thing to do.